Subject: [PATCH] newsfeed
---
Index: .idea/workspace.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/workspace.xml b/.idea/workspace.xml
--- a/.idea/workspace.xml	(revision 18640f31c85136055c5bf0ab9eceabfd986d84e6)
+++ b/.idea/workspace.xml	(revision b6f7f7c9e9756db8874ae37afa9f1295d2266551)
@@ -5,7 +5,6 @@
   </component>
   <component name="ChangeListManager">
     <list default="true" id="92677742-47d5-4b3c-958d-aa27adc18e6a" name="Changes" comment="">
-      <change beforePath="$PROJECT_DIR$/.idea/workspace.xml" beforeDir="false" afterPath="$PROJECT_DIR$/.idea/workspace.xml" afterDir="false" />
       <change beforePath="$PROJECT_DIR$/src/main/java/org/example/NewsFeedSolutionAlternative.java" beforeDir="false" afterPath="$PROJECT_DIR$/src/main/java/org/example/NewsFeedSolutionAlternative.java" afterDir="false" />
     </list>
     <option name="SHOW_DIALOG" value="false" />
Index: src/main/java/org/example/NewsFeedSolutionAlternative.java
===================================================================
diff --git a/src/main/java/org/example/NewsFeedSolutionAlternative.java b/src/main/java/org/example/NewsFeedSolutionAlternative.java
--- a/src/main/java/org/example/NewsFeedSolutionAlternative.java	(revision 18640f31c85136055c5bf0ab9eceabfd986d84e6)
+++ b/src/main/java/org/example/NewsFeedSolutionAlternative.java	(revision b6f7f7c9e9756db8874ae37afa9f1295d2266551)
@@ -78,6 +78,7 @@
 
         NewFeedAlternative newsfeed = new NewFeedAlternative();
         String[] arr;
+        int count=0;
 
         while (totalNumberOfRequests-- > 0) {
             arr = scanner.nextLine().split(" ");
@@ -89,8 +90,14 @@
                     List<NewsFeedEntityAlternative> res = (arr.length == 3) ?
                             newsfeed.getLatestNewsfeedEntities(Integer.parseInt(arr[1]), arr[2]) :
                             newsfeed.getLatestNewsfeedEntities(Integer.parseInt(arr[1]));
-                    for (NewsFeedEntityAlternative entity : res) {
-                        out.print((entity != null) ? entity.id + " " : "null ");
+
+                    int k = Integer.parseInt(arr[1]);
+                    for (int i = 0; i < k; i++) {
+                        if (i < res.size()) {
+                            out.print(res.get(i).id + " ");
+                        } else {
+                            out.print("null ");
+                        }
                     }
                     out.println();
                     break;
